src/cider/type/plan/UnaryExpr.cpp

```diff
@@ -143,60 +174,174 @@ JITExprValue& UOper::codegenUminus(CodegenContext& context, Analyzer::Expr* oper
+  auto args_desc =
+      JITFunctionEmitDescriptor{.ret_type = getJITTypeTag(target_ti.get_type()),
+                                .params_vector = {str_val.get(), str_len.get()}};
+  auto dim_val = func.createLiteral(JITTypeTag::INT32, target_ti.get_dimension());
+  const auto target_type = target_ti.get_type();
+  switch (target_type) {
+    case kBOOLEAN:
+      func_call += "bool";
+      break;
+    case kBIGINT:
+      func_call += "bigint";
+      break;
+    case kINT:
+      func_call += "int";
+      break;
+    case kSMALLINT:
+      func_call += "smallint";
+      break;
+    case kTINYINT:
+      func_call += "tinyint";
+      break;
+    case kFLOAT:
+      func_call += "float";
+      break;
+    case kDOUBLE:
+      func_call += "double";
+      break;
+    case kTIME:
+      func_call += "time";
+      args_desc.params_vector.push_back(dim_val.get());
+      break;
+    case kTIMESTAMP:
+      func_call += "timestamp";
+      args_desc.params_vector.push_back(dim_val.get());
+      break;
+    case kDATE:
+      func_call += "date";
+      break;
+    default:
+      CIDER_THROW(CiderCompileException,
+                  "Unimplemented type for string cast to " + toString(target_type));
+  }
+  return func.emitRuntimeFunctionCall(func_call, args_desc);
+}
+
+JITExprValue& UOper::codegenCast(CodegenContext& context, Analyzer::Expr* operand) {
+  JITFunction& func = *context.getJITFunction();
+  const auto& target_ti = get_type_info();
+  const auto& operand_ti = operand->get_type_info();
+  if (operand_ti.is_string() && target_ti.is_string()) {
+    // only supports casting from varchar to varchar
+    VarSizeJITExprValue operand_val(operand->codegen(context));
+    return set_expr_value(
+        operand_val.getNull(), operand_val.getLength(), operand_val.getValue());
+  } else if (target_ti.is_string()) {
+    FixSizeJITExprValue operand_val(operand->codegen(context));
+    auto string_heap_ptr = func.emitRuntimeFunctionCall(
+        "get_query_context_string_heap_ptr",
+        JITFunctionEmitDescriptor{.ret_type = JITTypeTag::POINTER,
+                                  .ret_sub_type = JITTypeTag::INT8,
+                                  .params_vector = {func.getArgument(0).get()}});
+    JITValuePointer ptr_and_len = codegenCastNumericToString(
+        context, operand_val.getValue(), string_heap_ptr, operand_ti);
+    auto ret_ptr = func.emitRuntimeFunctionCall(
+        "extract_str_ptr",
+        JITFunctionEmitDescriptor{.ret_type = JITTypeTag::POINTER,
+                                  .params_vector = {ptr_and_len.get()}});
+    auto ret_len = func.emitRuntimeFunctionCall(
+        "extract_str_len",
+        JITFunctionEmitDescriptor{.ret_type = JITTypeTag::INT32,
+                                  .params_vector = {ptr_and_len.get()}});
+    return set_expr_value(operand_val.getNull(), ret_len, ret_ptr);
+  } else if (operand_ti.is_string()) {
+    auto str_val = VarSizeJITExprValue(operand->codegen(context));
```

Comment:
need do a check on `target_ti` as it should be numeric?

Reply:
defined switch-case check in `codegenCastNumericToString`, not supported type will throw exception.